### PR TITLE
created validation for login, added api operation names to login and …

### PR DESCRIPTION
**Pull request by AinaArd**
…sign up methods

### CODE REVIEW

File: src/main/java/ru/itis/services/UserService.java [hunk 1]
```diff
@@ -79,6 +79,10 @@ public TokenDto login(UserForm userForm) {
         return usersRepository.findByLogin(userLogin);
     }
 
+    public boolean checkIfExists(String value) {
```
Comment (Dimanaux) [inline src/main/java/ru/itis/services/UserService.java]: SRP is violated in this class.

File: src/main/java/ru/itis/utils/UniqueLoginValidator.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import ru.itis.services.UserService;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class UniqueLoginValidator implements ConstraintValidator<UniqueLogin, String> {
+
+   private UserService userService;
```
[the remark this answers is not in the document]
Comment (AinaArd) [reply src/main/java/ru/itis/utils/UniqueLoginValidator.java]: fixed

File: src/main/java/ru/itis/services/UserService.java (hunk 1 above)
Comment (AinaArd) [reply src/main/java/ru/itis/services/UserService.java]: fixed

File: src/main/java/ru/itis/utils/UniqueLoginValidator.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+   private UsersRepository usersRepository;
+
+   @Autowired
+   public UniqueLoginValidator(UsersRepository usersRepository) {
+      this.usersRepository = usersRepository;
+   }
+
+   public boolean isValid(String value, ConstraintValidatorContext context) {
+      return value != null && !usersRepository.findByLogin(value).isPresent();   }
```
Comment (Dimanaux) [inline src/main/java/ru/itis/utils/UniqueLoginValidator.java]: ctrl alt L plz, add pmd to ci

Comment (AinaArd) [reply src/main/java/ru/itis/utils/UniqueLoginValidator.java]: fixed

**Pull request merged**